### DISCUSSION TITLE
What is __ASSET_JOB?

### DISCUSSION THREAD

**Discussion by marcilj** [dagster-io/dagster#18819] [Q&A] [answered]
I'm trying to build some test that compare the number of jobs defined in my project and the number of jobs in my project definition, and my assertion is failing because of a job called `__ASSET_JOB`.

Since I don't have any job called like this, can someone tell me, what is this job,? What does it do? Can I get rid of it?

Thank you

**Comment by smackesey** [comment, marked as answer]
`__ASSET_JOB` is an implementation detail. All runs in Dagster (currently) have an associated job-- when you select some assets in the UI outside of any job and hit "Materialize", `__ASSET_JOB` is used under the hood as the associated job. Sometimes there is more than one such job (named `__ASSET_JOB_<N>`) if there are multiple partitions definitions for assets.

You can't get rid of it, but you should be able to safely exclude any job name starting with `__ASSET_JOB` in your tests.
